lms/djangoapps/certificates/api.py

```diff
@@ -273,6 +274,27 @@ def set_cert_generation_enabled(course_key, is_enabled):
         log.info(u"Disabled self-generated certificates for course '%s'.", unicode(course_key))
 
 
+def is_certificate_invalid(student, course_key):
+    """Check that whether the student in the course has been invalidated
+    for receiving certificates.
+
+    Arguments:
+        student (user object): logged-in user
+        course_key (CourseKey): The course identifier.
+
+    Returns:
+        Boolean denoting whether the student in the course is invalidated
+        to receive certificates
+    """
+    is_invalid = False
+    certificate = GeneratedCertificate.certificate_for_student(student, course_key)
+    if certificate and certificate.status == CertificateStatuses.unavailable:
+        # in case of status 'un-available', check that if this certificate is invalidated
```

Comment:
Why do we check certificate invalidation only in the case of status `unavailable`? Shouldn't we check the certificate invalidation regardless of certificate status? Like even if the status of the certificate is "downloadable", but then the course instructor invalidated the student's certificate, we should return certificate invalidated, right? Let me know if this is not the case.